spec/cluster_getter_spec.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "cluster_getter"
+
+RSpec.describe ClusterGetter do
+  let(:ocn1) { 5 }
+  let(:ocn2) { 6 }
+  let(:ocn3) { 7 }
+  let(:ocn4) { 8 }
+  let(:ht) { build(:ht_item).to_hash }
+
+  before(:each) do
+    Cluster.create_indexes
+    Cluster.collection.find.delete_many
+  end
+
+  context "when merging two clusters" do
+    let(:c1) { create(:cluster, ocns: [ocn1]) }
+    let(:c2) { create(:cluster, ocns: [ocn2]) }
+    let(:htitem1) { build(:ht_item, ocns: [ocn1]).to_hash }
+    let(:htitem2) { build(:ht_item, ocns: [ocn2]).to_hash }
+    let(:holding1) { build(:holding, ocn: ocn1).attributes }
+    let(:holding2) { build(:holding, ocn: ocn2).attributes }
+    let(:ocn_resolution1) { build(:ocn_resolution, resolved: ocn1, deprecated: ocn3).attributes }
+    let(:ocn_resolution2) { build(:ocn_resolution, resolved: ocn2, deprecated: ocn4).attributes }
+
+    let(:merged_cluster) { described_class.new([ocn1, ocn2]).get }
```

Comment:
These bother me a little bit, because they generate clusters without "glue", i.e. htitem1 and htitem2 shouldn't be in the same cluster. On the other hand, part of the test is that it's not ClusterGetter's responsibility to manage cluster integrity, so maybe I should get over it.

Reply:
Right -- what would happen "for real" is that if you went to add that glue (either an htitem or ocn resolution containing both ocn1 and ocn2), `ClusterGetter` would first merge the clusters for ocn1 and ocn2, and the actual glue would be added after the cluster was changed. If the cluster gets merged, that happens inside a transaction, and the "glue" should get added within the scope of that transaction (so if it fails / gets rolled back, the cluster merge should get rolled back as well). We could add a test for that if we feel it would be helpful.